### DISCUSSION TITLE
Looping and Sending Mails does not work

### DISCUSSION THREAD

**Discussion by silvanhornstein** [Budibase/budibase#8191] [Help] [answered]
Hi Team!

Thanks for your great product and your efforts in supporting your community <3

I am struggling with the following problem: I have an automation that
1. Filters rows
2. Loops through this rows
3. Sends mails to the email adress in each of the row, with the HTML mail content containing a link ending with the id of the row.

However, the following unexpected behaviour is happening: The mail just contains the id of the row, but nothing of the other text I added to the HTML Content.

Any advice? Is this a bug or a mistake from my side?
Best Wishes!

**Comment by melohagan** [comment]
Hey @silvanhornstein 

Thanks for the nice words!

Would you be able to share some screenshots of what you have in your automation please?

**Reply by silvanhornstein** [reply]
Hey @melohagan Thanks for the quick turnaround : )

1. 
<img width="398" alt="image" src="https://user-images.githubusercontent.com/44234040/194898570-04a5725a-7302-4862-b1fa-7cd97451c2da.png">
(cron trigger -> queriing rows)

2. 
<img width="377" alt="image" src="https://user-images.githubusercontent.com/44234040/194899304-0d26dadd-f3d4-42ca-8b82-564b1377b7f2.png">
Looping through rows, sending mails to adresses in rows.

3. result:
Mail arrives, but just includes the _id, but not the remaing part of the html 
<img width="1018" alt="image" src="https://user-images.githubusercontent.com/44234040/194899763-dbb44ed9-32a2-46cd-aed6-cc856199a33a.png">


Thanks for your time and thoughts!!!!

**Comment by melohagan** [comment, marked as answer]
Thank you for the extra info @silvanhornstein 

Unfortunately this appears to be a bug specifically around the looping functionality. I've raised an issue here: https://github.com/Budibase/budibase/issues/8202

**Reply by radagarst** [reply]
Hey @melohagan
we are currently struggling with the same issue. Is there any estimate how long it might take for this to be fixed?

**Reply by melohagan** [reply]
Hey @radagarst I have a PR out for this one, hopefully this will be released reasonably soon. You can keep an eye on it here: https://github.com/Budibase/budibase/pull/8394
